icu4j/main/core/src/main/java/com/ibm/icu/util/IslamicCalendar.java

```diff
@@ -844,84 +1011,20 @@ protected int handleGetExtendedYear() {
-        long days = julianDay - CIVIL_EPOC;
-
-        if (cType == CalculationType.ISLAMIC_CIVIL || cType == CalculationType.ISLAMIC_TBLA) {
-            if (cType == CalculationType.ISLAMIC_TBLA) {
-                days = julianDay - ASTRONOMICAL_EPOC;
-            }
-            // Use the civil calendar approximation, which is just arithmetic
-            year  = (int)Math.floor( (30 * days + 10646) / 10631.0 );
-            month = (int)Math.ceil((days - 29 - yearStart(year)) / 29.5 );
-            month = Math.min(month, 11);
-        } else if (cType == CalculationType.ISLAMIC){
-            // Guess at the number of elapsed full months since the epoch
-            int months = (int)Math.floor(days / CalendarAstronomer.SYNODIC_MONTH);
-
-            monthStart = (long)Math.floor(months * CalendarAstronomer.SYNODIC_MONTH - 1);
-
-            if ( days - monthStart >= 25 && moonAge(internalGetTimeInMillis()) > 0) {
-                // If we're near the end of the month, assume next month and search backwards
-                months++;
-            }
-
-            // Find out the last time that the new moon was actually visible at this longitude
-            // This returns midnight the night that the moon was visible at sunset.
-            while ((monthStart = trueMonthStart(months)) > days) {
-                // If it was after the date in question, back up a month and try again
-                months--;
-            }
-
-            year = months >=  0 ? ((months / 12) + 1) : ((months + 1 ) / 12);
-            month = ((months % 12) + 12 ) % 12;
-        } else if (cType == CalculationType.ISLAMIC_UMALQURA) {
-            long umalquraStartdays = yearStart(UMALQURA_YEAR_START);
-            if( days < umalquraStartdays) {
-                // Use Civil calculation
-                year  = (int)Math.floor( (30 * days + 10646) / 10631.0 );
-                month = (int)Math.ceil((days - 29 - yearStart(year)) / 29.5 );
-                month = Math.min(month, 11);
-            } else {
-                int y =UMALQURA_YEAR_START-1, m =0;
-                long d = 1;
-                while(d > 0) {
-                    y++;
-                    d = days - yearStart(y) +1;
-                    if(d == handleGetYearLength(y)) {
-                        m=11;
-                        break;
-                    } else if(d < handleGetYearLength(y) ) {
-                        int monthLen = handleGetMonthLength(y, m);
-                        m=0;
-                        while(d > monthLen) {
-                            d -= monthLen;
-                            m++;
-                            monthLen = handleGetMonthLength(y, m);
-                        }
-                        break;
-                    }
-                }
-                year = y;
-                month = m;
-            }
+        switch (cType) {
+            case ISLAMIC_CIVIL:
+                handleComputeFieldsIslamicCivil(julianDay - CIVIL_EPOC);
```

Comment:
If these symbols are internal and you can change them, the word is spelled "epoch".  `CIVIL_EPOCH` and `ASTRONOMICAL_EPOCH`.